### PR TITLE
[ng] glossary update

### PR DESCRIPTION
**Pull request by chalin**
- Mainly a sync with ng.io.
- Re-introduces the AOT and JIT entries.
- Other copyedits.

As usual, you can ignore the cache and patch files.

### CODE REVIEW

File: src/angular/glossary.jade [hunk 1]
```diff
@@ -205,18 +217,20 @@ a#decoration
...
+
 :marked
   ## Directive
 .l-sub-section
   :marked
     An Angular class responsible for creating, reshaping, and interacting with HTML elements
-    in the browser DOM. Directives are Angular's most fundamental feature.
+    in the browser DOM. The directive is Angular's most fundamental feature.
+
+    A directive is ususally associated with an HTML element or attribute.
```
Comment (kwalrath) [inline src/angular/glossary.jade]: ususally -> usually

File: src/angular/glossary.jade [hunk 2]
```diff
@@ -425,12 +450,11 @@ a#Q
     and performing other similar actions that cause the application to
     replace one view with another.
 
-    The Angular [component router](!{docsLatest}/guide/router.html) is a richly featured mechanism for configuring and managing the entire view navigation process including the creation and destruction
+    The Angular component router is a richly featured mechanism for configuring and managing the entire view navigation process, including the creation and destruction
```
Comment (kwalrath) [inline src/angular/glossary.jade]: Why remove the link to the router docs?

**Comment by kwalrath** [comment]
I noticed a few small issues, but nothing worth holding back this PR for.

**Comment by kwalrath** [comment]
@kapunahelewong would it be easy for you to make these fixes in the TS docs, or has that ship sailed?

**Comment by chalin** [comment]
The TS docs, in their Jade form, haven't been frozen yet. Even after Jade files are converted to markdown, making updates should still be possible :).

File: src/angular/glossary.jade [hunk 3]
```diff
@@ -51,36 +61,37 @@ include _util-fns
...
-    Almost always refers to [Data Binding](#data-binding) and the act of
+    Usually refers to [data binding](#data-binding) and the act of
     binding an HTML object property to a data object property.
 
-    May refer to a [dependency injection](#dependency-injection) binding
-    between a "token", also referred to as a "key", and a dependency [provider](#provider).
-    This more rare usage should be clear in context.
+    Sometimes refers to a [dependency-injection](#dependency-injection) binding
+    between a "token"&mdash;also referred to as a "key"&mdash;and a dependency [provider](#provider).
+    When using this more rare usage, be clear in context.
```
Comment (chalin) [inline src/angular/glossary.jade]: Ah, hand't noticed that. I've just removed it for now.

File: src/angular/glossary.jade (hunk 1 above)
Comment (chalin) [inline src/angular/glossary.jade]: fixed

File: src/angular/glossary.jade (hunk 2 above)
Comment (chalin) [inline src/angular/glossary.jade]: I don't know why it was removed from the TS side. I decided to match the TS entry ... though I did hesitate. @kapunahelewong can you find out why the link was removed?

**Comment by chalin** [comment]
Travis reports green and @kwalrath has approved so I'm merging.
Also, I propagated back the changes to the TS docs.

File: src/angular/glossary.jade (hunk 2 above)
Comment (kapunahelewong) [inline src/angular/glossary.jade]: @kwalrath @chalin Sure, I can investigate. Seems like such a particular choice. I'll dig and let you two know what I find.

**Comment by kapunahelewong** [comment]
@chalin @kwalrath The v.2 docs are frozen but I can still submit PRs for the v.4 jade docs.

**Comment by chalin** [comment]
Right, v4 docs it is.

**Pull request merged**